.github/actions/setup-node-if-needed/action.yml

```diff
@@ -15,7 +15,7 @@ runs:
       id: check-node
       shell: bash
       run: |
-        if command -v node >/dev/null 2>&1; then
+        if node --version 2>/dev/null | grep -q "v${{ inputs.node_version }}"; then
```

Comment:
This pre-existing check would resolve true if the installed version of node was different to what was specified via `inputs.node_version`. Harden it a bit by verifying that the actual installed version of node reports something that gives a strong hint at being the right version. This _should_ be compatible with specifying additional constraints on the version - i.e. minor and patch levels.

I wonder if there's a cleaner way to do this - i.e. I'd suspect `setup-node` has some built-in version checker, but since action was only recently by @Reinis-FRP I'm assuming there was some blocker there.